### PR TITLE
add BUILD_METADATA fallback when parsing

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR adds a fallback to empty build metadata when BUILD_METADATA contains invalid JSON.

Example `warning` log for `BUILD_METADATA={...}`:

```
20:57:57.872 [warning] failed to parse $BUILD_METADATA, reason: ** (Jason.DecodeError) unexpected byte at position 1: 0x2E (".")
```

Fixes https://github.com/plausible/analytics/issues/2491

### Tests
- [x] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -95,7 +96,16 @@ ch_db_url =
...
-  |> Jason.decode!()
+  |> Jason.decode()
+  |> case do
+    {:ok, build_metadata} ->
+      build_metadata
+
+    {:error, error} ->
+      error = Exception.format(:error, error)
+      Logger.warn("failed to parse $BUILD_METADATA, reason: " <> error)
+      _fallback = %{}
```
Comment (ruslandoga) [inline config/runtime.exs:107]: Should empty fallback be mentioned in the warning?

Comment (ruslandoga) [reply config/runtime.exs:107]: Should `$BUILD_METADATA` contents be added to the warning?

Comment (ukutaht) [reply config/runtime.exs:107]: Yes to both :)

Comment (ruslandoga) [reply config/runtime.exs:107]: The warning now looks like this

```
21:33:27.449 [warning] failed to parse $BUILD_METADATA: ** (Jason.DecodeError) unexpected byte at position 1: 0x2E (".")

    $BUILD_METADATA was set to "{...}"

    falling back to empty value, as if `BUILD_METADATA={}`
```

Comment (ruslandoga) [reply config/runtime.exs:114]: Screenshot 2022-12-05 at 21 35 11

I added whitespace to make it stand out. Not sure if it makes sense though...

Comment (ruslandoga) [reply config/runtime.exs:114]: With the real JSON it'd be

```
21:38:33.996 [warning] failed to parse $BUILD_METADATA: ** (Jason.DecodeError) unexpected byte at position 38: 0x70 ("p")

    $BUILD_METADATA was set to "{\"tags\":[\"plausible/analytics:v1.5.0\",plausible/analytics:v1.5,plausible/analytics:v1,\"plausible/analytics:latest\"],\"labels\":{\"org.opencontainers.image.title\":\"analytics\",\"org.opencontainers.image.description\":\"Simple, open-source, lightweight (< 1 KB) and privacy-friendly web analytics alternative to Google Analytics.\",\"org.opencontainers.image.url\":\"https://github.com/plausible/analytics\",\"org.opencontainers.image.source\":\"https://github.com/plausible/analytics\",\"org.opencontainers.image.version\":\"v1.5.0\",\"org.opencontainers.image.created\":\"2022-12-02T15:45:09.831Z\",\"org.opencontainers.image.revision\":\"f9ab2650b3192021d5808cc8f20064aa8af543f4\",\"org.opencontainers.image.licenses\":\"AGPL-3.0\"}}"

    falling back to empty value, as if `BUILD_METADATA={}`
```

**Review by ukutaht — APPROVED** [review]

**Pull request merged**